### ISSUE TITLE
Concating array

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1445]
Description (screenshot):
--
![Screenshot from 2019-11-08 09-55-15](https://user-images.githubusercontent.com/866668/68463021-0ed72580-020e-11ea-9103-01956f161362.png)

```php
    /** @var string[] $a */
    $a = ['a'];
    $b = implode('', $a);
    echo "test {$a}";
    echo 'test' . $a;
```

PHPStorm didn't warn me in a case like that, including a array in the string, it didn't even mark the 2nd var as unused. (think unused never triggers in global space, just in functions).

If i mark it with `/** @var string[] $a */`, it triggers `PhpToStringImplementationInspection`,
but if the type is array, thats not triggered.

Is it just a bug in PS2019.3 EAP (193.5096.23),
or have it never shown a warning for concating arrays?

Sugestion: An inspection that warns if concatination is done on arrays.

**Comment by kalessil** [comment]
Thank you for reporting @puggan. I'm trying to not duplicate much PhpStorm functionality and would like to encorage you reporting this case as a bug in PhpToStringImplementationInspection, so PhpStrom team could care. Will this work for you?

**Comment by puggan** [comment]
https://youtrack.jetbrains.com/issue/WI-49793

**Comment by kalessil** [comment]
Super, thank you very much!
